Add Purpose/Vision section to GTR (#2261)
cncf/toc: toc_subprojects/project-reviews-subproject/general-technical-questions.md

- **Project Version:** 
- **Website:**
- **Date Updated:** YYYY-MM-DD
- **Template Version:** v1.0.1
- **Description:** <!-- Short project description --> 

## Day 0 - Planning Phase

### Scope

>   * What technical and business requirements and problems is the project intended to address? How does it address them?

Reviewer: ```suggestion
  * Describe the project's vision and goals.
```

>   * What unique values does the project offer compared to similar and related projects?

Reviewer: This is addressed in the application itself

>   * What are the primary use cases for the project? What additional use cases are supported by the project?

Reviewer: ```suggestion
  * Describe the primary use cases for the project. Describe any additional use cases the project supports. 
```

>   * Are these use cases that are explicitly unsupported by the project and why?

Reviewer: This line is covered by the 'out of scope' question

> - **Template Version:** v1.0

Reviewer 2: Likely we can keep this version and the top-level version in sync?

Author: Set them both at `v1.0.1`, thanks!

- **Description:** <!-- Short project description --> 

> ## Day -1 - Purpose and Vision

Reviewer 2: I don't feel strongly but I am curious if `Day -1` is the most clear versus something like:

`## Project Definition - Purpose, Value, and Use Cases`

Author: That makes sense. I merged the two new questions into "Day 0", WDYT?
